internal/zinc-compile-core/src/test/scala/sbt/internal/inc/javac/JavaCompilerSpec.scala

```diff
@@ -82,9 +90,8 @@ class JavaCompilerSpec extends UnitSpec with DiagrammedAssertions {
   }
 
   def findsDocErrors(compiler: XJavaTools) = IO.withTemporaryDirectory { out =>
-    val (result, problems) =
+    val (_, problems) =
       doc(compiler, Seq(knownSampleErrorFile), Seq("-d", out.getAbsolutePath))
-    assert(result)
```

Comment:
What's the motivation for this change?

Reply:
`doc(...)` correctly returns `false` here so the assert fails. Maybe that's also a JDK dependent behavior?

Reply:
Maybe we can also show that in the test, as in the case of the error mismatch?

Reply:
```scala
    // exit code for `javadoc` commandline is JDK dependent
    assert(result == {
      sys.props("java.specification.version") match {
        case "1.8" | "9" => true
        case _           => false
      }
    })
```